docusaurus/docs/create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins.md

```diff
@@ -0,0 +1,486 @@
+---
+id: add-authentication-for-data-source-plugins
+title: Add authentication for data source plugins
+sidebar_position: 2
+description: How to add authentication for data source plugins.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - authentication
+  - data source
+  - datasource
+---
+
+# Add authentication for data source plugins
+
+Grafana plugins can perform authenticated requests against a third-party API by using the _data source proxy_ or through a custom a _backend plugin_.
+
+## Choose an authentication method
+
+Configure your data source plugin to authenticate against a third-party API in one of either of two ways:
+
+- Use the [_data source proxy_](#authenticate-using-the-data-source-proxy) method, or
+- Build a [_backend plugin_](#authenticate-using-a-backend-plugin).
+
+| Case                                                                                            | Use                             |
+| ----------------------------------------------------------------------------------------------- | ------------------------------- |
+| Do you need to authenticate your plugin using Basic Auth or API keys?                           | Use the data source proxy.      |
+| Does your API support OAuth 2.0 using client credentials?                                       | Use the data source proxy.      |
+| Does your API use a custom authentication method that isn't supported by the data source proxy? | Use a backend plugin.           |
+| Does your API communicate over a protocol other than HTTP?                                      | Build and use a backend plugin. |
+| Does your plugin require alerting support?                                                      | Build and use a backend plugin. |
+
+## Encrypt data source configuration
+
+Data source plugins have two ways of storing custom configuration: `jsonData` and `secureJsonData`.
+
+Users with the Viewer role can access data source configuration such as the contents of `jsonData` in cleartext. If you've enabled anonymous access, anyone who can access Grafana in their browser can see the contents of `jsonData`.
+
+Users of [Grafana Enterprise](https://grafana.com/products/enterprise/grafana/) can restrict access to data sources to specific users and teams. For more information, refer to [Data source permissions](https://grafana.com/docs/grafana/latest/enterprise/datasource_permissions).
+
+:::important
```

Comment:
<img width="877" alt="Screenshot 2023-08-16 at 17 27 49" src="https://github.com/grafana/plugin-tools/assets/19860021/8f75661d-ec36-45cd-ae90-f55a8f0711ec">

Important just renders as info. This doesn't feel appropriate, maybe caution would be better?
<img width="392" alt="Screenshot 2023-08-16 at 17 29 15" src="https://github.com/grafana/plugin-tools/assets/19860021/7803902d-a38b-4c53-babf-a8c62de8ff78">

Reply:
I'm also not a fan of this section having two admonitions directly next to each other. We could combine the warning about jsonData with the explanation (second sentence in this section)?